src/App.tsx

```diff
@@ -1,222 +1,203 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  // useMemo,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import {
+  getTodos,
+  createTodo,
+  removeTodo,
+  changeTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { FilterCondition, ErorrMessage } from './types/enums';
+import { Header } from './components/Main/Header';
+import { TodoList } from './components/Main/TodoList';
+import { Footer } from './components/Main/Footer';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todosList, setTodosList] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [errorText, setErrorText] = useState('');
+  const [filterCondition, setFilterCondition] = useState(FilterCondition.ALL);
+  const [isAdding, setIsAdding] = useState(false);
+  const [tempNewTask, setTempNewTask] = useState<Todo | null>(null);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+  const [updatingTodoIds, setUpdatingTodoIds] = useState<number[]>([]);
+
+  if (isError) {
+    setTimeout(() => setIsError(false), 3000);
+  }
+
+  const getSelected = useCallback(
+    (allTodos: Todo[]): Todo[] => {
+      switch (filterCondition) {
+        case FilterCondition.COMPLETED:
+          return allTodos.filter(item => item.completed === true);
+
+        case FilterCondition.ACTIVE:
+          return allTodos.filter(item => item.completed === false);
+
+        case FilterCondition.ALL:
+        default:
+          return allTodos;
+      }
+    }, [filterCondition],
+  );
+
+  const addTodo = useCallback(async (todoData: Omit<Todo, 'id'>) => {
+    const fullTodoData = { ...todoData, userId: user?.id };
+
+    try {
+      const temporaryTodo = {
+        ...todoData,
+        id: 0,
+      };
+
+      setTempNewTask(temporaryTodo);
+
+      setIsAdding(true);
+      const newTodo = await createTodo(fullTodoData);
+
+      setTodosList(currentTodos => [...currentTodos, newTodo]);
+    } catch {
+      setErrorText(ErorrMessage.ON_ADD);
+    } finally {
```

Comment:
U forgot to add setIsError, let's create `showError` function to avoid code repeating and errors like that

Reply:
+